name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/ChartToolsManager.java

```diff
@@ -0,0 +1,523 @@
+                            .atZone(ZoneId.systemDefault()).toLocalDate();
+        }
+
+        private void drawCrosshairDateTextbox(PaintEvent e, Point p)
+        {
+            LocalDate date = getDateTime(p);
+            String xText = Values.Date.format(date);
+
+            // Add margin to text
+            Point txtXExtend = e.gc.textExtent(xText);
+            txtXExtend.x += 5;
+            txtXExtend.y += 5;
+
+            Point rectPoint = new Point(0, e.height - txtXExtend.y - 2);
+            Point textPoint = new Point(0, e.height - txtXExtend.y + 1);
+
+            // Visual shift vertical of the container
+            if (p.x <= e.width / 2)
+            {
+                rectPoint.x = p.x + 5;
+                textPoint.x = p.x + 8;
+            }
+            else
+            {
+                rectPoint.x = p.x - txtXExtend.x - 5;
+                textPoint.x = p.x - txtXExtend.x - 1;
+            }
+
+            e.gc.setBackground(Colors.brighter(color));
+            e.gc.setForeground(Colors.getTextColor(color));
+
+            e.gc.fillRoundRectangle(rectPoint.x, rectPoint.y, txtXExtend.x, txtXExtend.y, PADDING, PADDING);
+            e.gc.drawText(xText, textPoint.x, textPoint.y, true);
+        }
+
+        private double getYValue(Point p)
+        {
+            return chart.getAxisSet().getYAxis(0).getDataCoordinate(p.y);
+        }
+
+        private void drawCrosshairValueTextbox(PaintEvent e, Point p)
+        {
+            double value = getYValue(p);
+
+            String yText = new DecimalFormat(Values.Quote.pattern()).format(value);
+
+            // Add margin to text
+            Point txtYExtend = e.gc.textExtent(yText);
+            txtYExtend.x += 5;
+            txtYExtend.y += 5;
+
+            Point rectPoint = new Point(e.width - txtYExtend.x - 2, 0);
+            Point textPoint = new Point(e.width - txtYExtend.x + 1, 0);
+
+            // Visual shift horizontally of the container
+            if (p.y <= e.height / 2)
+            {
+                rectPoint.y = p.y + 4;
+                textPoint.y = p.y + 7;
+            }
+            else
+            {
+                rectPoint.y = p.y - txtYExtend.y - 4;
+                textPoint.y = p.y - txtYExtend.y - 1;
+            }
+
+            e.gc.setBackground(Colors.brighter(color));
+            e.gc.setForeground(Colors.getTextColor(color));
+
+            e.gc.fillRoundRectangle(rectPoint.x, rectPoint.y, txtYExtend.x, txtYExtend.y, PADDING, PADDING);
+            e.gc.drawText(yText, textPoint.x, textPoint.y, true);
+        }
+
+        private void drawCrosshairValueSecondAxisTextbox(PaintEvent e, Point p)
+        {
+            IAxis axis = chart.getAxisSet().getYAxis(2);
+            if (!axis.getTick().isVisible())
+                return;
+
+            double value = axis.getDataCoordinate(p.y);
+
+            String yText = new DecimalFormat("+#.##%;-#.##%").format(value); //$NON-NLS-1$
```

Comment:
Maybe it is better if we get formattings from here (analogous to L487)
```suggestion
            String yText = Values.PercentWithSign.format(value);
```